### DISCUSSION TITLE
Consistent defaultValues without setting extra variables on init

### DISCUSSION THREAD

**Discussion by TheMeteoRain** [react-hook-form/react-hook-form#3022] [Q&A] [answered]
Hey, I had a problem when using register function manually. I tried to set defaultValues, but the result was always empty object when calling getValues function. Ultimately the problem was that the shouldUnregister was on default true. Once I set it to false, I could see my defaultValues normally. This took me good hour or so to figure out. 

For those who come after me. Is there any kind of safe measurement you could implement for this use case? I would even suggest to set the default so false. So it does not infer with new setups or those who upgrade from 5 -> 6 (this is my case) and rather leave it as an option for those who want to unregister components. Of course you could invert what I just said and argue that it's better as it is. Though I really think it just confuses those who want to use manual register, since there is no documentation that if you want to use manual register -> remember to set this variable to false.

> It doesn't auto populate with the manually registered input (eg: register({ name: 'test' })) because the custom register field does not provide the ref to React Hook Form.

This was the only documentation that even comes near to the problem, but it only talk about the automatic value population. Which is not the topic of this problem.

**Comment by bluebill1049** [comment]
provide a codesandbox for what you describing above.

**Comment by TheMeteoRain** [comment, marked as answer]
Here is the codesandbox: https://codesandbox.io/s/react-hook-form-basic-forked-qlr4j?file=/src/index.js:0-3203

I don't know why react-hook-form latest version did not show me that it has shouldUnregister variable, but if the manual register form had that as false, then we could see inputs getting populated also.

**Reply by bluebill1049** [reply]
`getVlaues()` **doesn't** return defaultValues

**Reply by TheMeteoRain** [reply]
Yes, getValues() is the current data of the form, but I can say that it does return defaultValues if I haven't entered any data to form. As in, it does represent the data that was passed to defaultValues. But I'm not here to talk about semantics 🤔

**Reply by bluebill1049** [reply]
yes, I am not talking about semantics either, we have tried in one of the patches and it creates too many side effects and bugs. If you want to get `defautValues`, store that in context or use inject into FormProvider (if you are not using Typescript)

**Reply by TheMeteoRain** [reply]
Okay, I think I might have wrongly explained the problem. The issue ain't that I want `defaultValues`, but the issue is that when manually registering components my `getValues()` is empty when I've had my `defaultValues` set.  You can see how this differs for manual and automatic register in the codesandbox. The other one give me the values with `getValues()` but the other does not.

**Reply by bluebill1049** [reply]
<img width="652" alt="Screen Shot 2020-09-28 at 9 30 50 pm" src="https://user-images.githubusercontent.com/10513364/94427109-ebc0b600-01d1-11eb-910f-14582026afac.png">

I am confused here, both return the same empty object.

**Reply by TheMeteoRain** [reply]
I don't know if this is just codesandbox things, but try to press ctrl+s. You can see that the other one has properties. 😕

**Reply by bluebill1049** [reply]
- getVlaues() doesn't return `defaultValues`
- with `register` second re-render will return form values, because inputs now have values.

conclusion `getValues()` **doesn't** return `defautlValues` and should only be relyed on reading **current** form values.

**Reply by TheMeteoRain** [reply]
I agree for all the above. Though can you explain why does not the same happen for manual registering?

**Reply by bluebill1049** [reply]
for the manual register, 

1. there are no inputs to be registered and set the input value
2. `setValue` will never invoke as well to set the values.

register works because the moment input rendered, they r based called the `setValue`.
